### ISSUE TITLE
Help making pull request: Ability to add `derives` clause

### ISSUE THREAD

**Issue by oscar-broman** [scalapb/ScalaPB#1584]
I might be able to create a pull request from this, but I'd need some guidance!

This is a local diff which achieves what I'm after (circe codecs for all types):
```
$ git diff
diff --git a/compiler-plugin/src/main/scala/scalapb/compiler/ProtobufGenerator.scala b/compiler-plugin/src/main/scala/scalapb/compiler/ProtobufGenerator.scala
index a484b7c2..7bf3d0ba 100644
--- a/compiler-plugin/src/main/scala/scalapb/compiler/ProtobufGenerator.scala
+++ b/compiler-plugin/src/main/scala/scalapb/compiler/ProtobufGenerator.scala
@@ -1353,7 +1353,7 @@ class ProtobufGenerator(
       .indent
       .indent
       .call(printConstructorFieldList(message))
-      .add(s") extends ${message.baseClasses.mkString(" with ")} {")
+      .add(s") extends ${message.baseClasses.mkString(" with ")} derives io.circe.Codec.AsObject {")
       .call(generateSerializedSizeForPackedFields(message))
       .call(generateSerializedSize(message))
       .call(generateWriteTo(message))
diff --git a/compiler-plugin/src/main/scala/scalapb/compiler/SealedOneofsGenerator.scala b/compiler-plugin/src/main/scala/scalapb/compiler/SealedOneofsGenerator.scala
index 98217311..b3ff3dee 100644
--- a/compiler-plugin/src/main/scala/scalapb/compiler/SealedOneofsGenerator.scala
+++ b/compiler-plugin/src/main/scala/scalapb/compiler/SealedOneofsGenerator.scala
@@ -43,7 +43,7 @@ class SealedOneofsGenerator(message: Descriptor, implicits: DescriptorImplicits)
         val sealedOneofUniversalMark = if (message.isUniversalTrait) "Any with " else ""

         fp.add(
-          s"sealed trait $sealedOneofName $bases{"
+          s"sealed trait $sealedOneofName $bases derives io.circe.Codec.AsObject {"
         ).addIndented(
           s"type MessageType = $baseType",
           s"final def isEmpty = this.isInstanceOf[${sealedOneofType}.Empty.type]",
@@ -56,7 +56,7 @@ class SealedOneofsGenerator(message: Descriptor, implicits: DescriptorImplicits)
           .indented(
             _.add(s"case object Empty extends $sealedOneofType", "")
               .add(
-                s"sealed trait $sealedOneofNonEmptyName extends $sealedOneofUniversalMark$sealedOneofType"
+                s"sealed trait $sealedOneofNonEmptyName extends $sealedOneofUniversalMark$sealedOneofType derives io.circe.Codec.AsObject"
               )
               .add(
                 s"def defaultInstance: ${sealedOneofType} = Empty",
```

How do I add a setting for this that could be specified both globally and per message?

**Comment by thesamet** [comment]
Hi Oscar, thanks for suggesting this feature and contributing the PR. I had started to work on this in parallel, looked at your code and made adjustments in mine as necessary.  One difference is that I refrained from introducing `target_regex` in `aux_message_options` but expanded `target` to accept `*` to indicate to apply to all messages in scope. Will this work for you?

I'd really appreciate before I cut a release if you will be able to try out the snapshot release `0.11.13+88-1d78ac52-SNAPSHOT` and see if it works for? Some documentation is at https://github.com/scalapb/ScalaPB/blob/master/CHANGELOG.md#01114 , and

**Comment by thesamet** [comment]
The previous commit has the diff you posted. Can you try again? I'm not sure why you were not able to use scalapb_runtime-3. Tthe new version is [published on sonatype](https://oss.sonatype.org/content/repositories/snapshots/com/thesamet/scalapb/scalapb-runtime_3/0.11.13+90-3eb3bf7e-SNAPSHOT/), but also the previous version did. Are your resolvers configured correctly for the project/

**Comment by thesamet** [comment]
Perfect. Will close the issue now and cut a release soon. Don't hesitate to start a new issue or comment here if anything left unresolved.
